File: digiflazz/src/main/java/id/indosw/digiflazz/api/sign/SignMaker.java
package id.indosw.digiflazz.api.sign;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class SignMaker {
    public static String encrypt(final String text) {
        try {
            // Create MD5 Hash
            MessageDigest digest = MessageDigest
                    .getInstance("MD5");
            digest.update(text.getBytes());
            byte[] messageDigest = digest.digest();

            // Create Hex String
            StringBuilder hexString = new StringBuilder();
            for (byte b : messageDigest) {
                StringBuilder h = new StringBuilder(Integer.toHexString(0xFF & b));
                while (h.length() < 2)
                    h.insert(0, "0");
                hexString.append(h);
            }
            return hexString.toString();

        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return "";
    }

    public static String getSign(String username, String key, String extensionSign){
        return encrypt(username + key + extensionSign);
    }
}

File: digiflazz/src/main/java/id/indosw/digiflazz/api/sign/SignMaker.kt
package id.indosw.digiflazz.api.sign

import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

object SignMaker {
    private fun encrypt(text: String): String {
        try {
            // Create MD5 Hash
            val digest = MessageDigest
                    .getInstance("MD5")
            digest.update(text.toByteArray())
            val messageDigest = digest.digest()

            // Create Hex String
            val hexString = StringBuilder()
            for (b in messageDigest) {
                val h = StringBuilder(Integer.toHexString(0xFF and b.toInt()))
                while (h.length < 2) h.insert(0, "0")
                hexString.append(h)
            }
            return hexString.toString()
        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
        }
        return ""
    }

    @JvmStatic
    fun getSign(username: String, key: String, extensionSign: String): String {
        return encrypt(username + key + extensionSign)
    }
}